### ISSUE TITLE
UCCRecalibration fit method running on one CPU core

### ISSUE THREAD

**Issue by AdelNamani** [IBM/UQ360#23]
`UCCRecalibration.fit` runs on only one CPU core.
Is there a way to parallelize the execution of this method?

**Comment by pronics2004** [comment]
@jirin1a do you have thoughts on this?

**Comment by jirin1a** [comment]
@AdelNamani, yes, the fit() method currently uses a single processor. Algorithmically, it can be parallelized. Before we look into this, can you give us more detail: what is the size of the data object with which you are invoking fit() and the amount of data you are recalibrating?

**Comment by AdelNamani** [comment]
Here are some details about the size of the data object I am trying to train on:
```
Int64Index: 229297 entries, 
Columns: 605 entries, 
dtypes: float64(84), int64(3), object(3), uint64(1), uint8(514)
memory usage: 273.3+ MB
```
This represents 25% of my train data.
Executing the `UCCRecalibration.fit` takes approximately 2 hours.

**Comment by jirin1a** [comment]
I see. We may put this request on our todo list - speeding up the fit() call make sense. 
In the meantime, to obtain better runtimes, subsampling the data used in the fit call may be a suitable alternative. From the instance naming, I am surmising that after the `fit()` call you are rescaling all your data to minimize a cost (?). If so, the data object, say `X`, can be  randomly subsampled to, say, 10% of the original size with the resulting optimum applied to the full batch of data.

**Comment by AdelNamani** [comment]
That's what I am doing actually, thanks!
